### PR TITLE
Implement ctrl-shift click to select multiple documents

### PR DESCRIPTION
**Pull request by languy**
[Preview this branch](https://cosmos-explorer-preview.azurewebsites.net/pull/1851)

This change implements the ability to select multiple documents by clicking with Ctrl and/or Shift. It will select/deselect individual items while keeping the other currently selected items, or select/deselect a range of documents. The behavior follows Windows File Explorer.

The mouse cursor is also changed to a pointer.

For single selection toggling, it's Command-click on Mac and Ctrl-Click on other platforms.

When selection is disabled (Fabric read-only), revert to just click behavior.

2024-05-30 shift ctrl selection

### CODE REVIEW

File: src/Explorer/Tabs/DocumentsTabV2/DocumentsTableComponent.tsx [hunk 1]
```diff
@@ -121,24 +123,58 @@ export const DocumentsTableComponent: React.FC<IDocumentsTableComponentProps> =
     [columnHeaders],
   );
 
-  const onIdClicked = useCallback((index: number) => onSelectedRowsChange(new Set([index])), [onSelectedRowsChange]);
+  const [selectionStartIndex, setSelectionStartIndex] = React.useState<number>(undefined);
+  const onTableCellClicked = useCallback(
+    (e: React.MouseEvent, index: number) => {
+      const result = selectionHelper(selectedRows as Set<number>, index, e.shiftKey, e.ctrlKey, selectionStartIndex);
```
Comment (sevoku) [inline src/Explorer/Tabs/DocumentsTabV2/DocumentsTableComponent.tsx:129]: For MacOS it should be `e.metaKey` instead of `e.ctrlKey` if I'm not mistaken.

**Review by sevoku — COMMENTED** [review]
LGTM works like a charm, but let's don't forget about MacOS.

File: src/Explorer/Tabs/DocumentsTabV2/DocumentsTableComponent.tsx (hunk 1 above)
Comment (languy) [reply src/Explorer/Tabs/DocumentsTabV2/DocumentsTableComponent.tsx]: That is correct. Turns out we already had code to detect platform and check ctrl vs meta for Tables API which I moved to a more common place.

**Review by sevoku — APPROVED** [review]
Nice, verified on MacOS as well.

**Review by jawelton74 — APPROVED** [review]

**Pull request merged**